.github/workflows/ci.yml

```diff
@@ -110,19 +110,19 @@ jobs:
       - name: Install Rust
         uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: nightly-2023-12-28
+          toolchain: nightly-2025-02-22
           components: rust-src
       - name: Install avr-gcc, binutils, and libc
         run: sudo apt-get install -y avr-libc binutils-avr gcc-avr
       - name: Compile board crate and examples
         if: "${{ matrix.m.type == 'board' && matrix.m.examples }}"
-        run: cd "examples/${{ matrix.m.name }}" && cargo build --bins
+        run: cd "examples/${{ matrix.m.name }}" && cargo build --release --bins
       - name: Compile board crate (without examples)
         if: "${{ matrix.m.type == 'board' && !matrix.m.examples }}"
-        run: cd "arduino-hal/" && cargo build --features "${{ matrix.m.name }}"
+        run: cd "arduino-hal/" && cargo build --release --features "${{ matrix.m.name }}"
       - name: Test-compile HAL crate for an MCU
         if: "${{ matrix.m.type == 'mcu' }}"
-        run: cd "mcu/${{ matrix.m.crate }}" && cargo build --features "${{ matrix.m.name }}" -Z build-std=core --target "../../avr-specs/avr-${{ matrix.m.spec }}.json"
+        run: cd "mcu/${{ matrix.m.crate }}" && cargo build --release --features "${{ matrix.m.name }}" -Z build-std=core
```

Comment:
shouldn't we have something like `--target=avr-none -C target-cpu={{matrix.something}}` in here?

Reply:
Yeah, with `avr-none` being upstream now, we should start making use of it everywhere.  I do wonder, however, what to do with the other target spec changes we currently do:

https://github.com/Rahix/avr-hal/blob/d1782c69236dcd4cb73dd1d7c414136cced86091/avr-specs/sync-from-upstream.py#L57-L64

https://github.com/Rahix/avr-hal/blob/d1782c69236dcd4cb73dd1d7c414136cced86091/avr-specs/sync-from-upstream.py#L103

Reply:
`max-atomic-width` is already set to 16:

https://github.com/rust-lang/rust/blob/ad27045c31a9f37ad7d44ca2a403de52d1a896d3/compiler/rustc_target/src/spec/targets/avr_none.rs#L25

I'm not sure on `no-default-libraries`, though - gotta look at it.

As for the extra link arguments, they can be always passed via `RUSTFLAGS` (e.g. through `.cargo/config`) or even through a `build.rs` script.

Reply:
I tried adding `-C default-linker-libraries` to Cargo.toml for one of the problematic cpus and while it removed the `-nodefaultlibs` argument from the `avr-gcc` invocation the linker error remains.